src/sentry/interfaces/security.py

```diff
@@ -354,6 +354,18 @@ def from_raw(cls, raw):
 
         # For CSP, the values we want are nested under the 'csp-report' key.
         raw = raw['csp-report']
+
+        if 'effective-directive' not in raw:
+            # For Firefox, parse effective-directive from violated-directive
+            # https://bugzil.la/1192684#c8
+            violated_directive = raw['violated-directive'].split(None, 1)
```

Comment:
Isn't this highly likely to be a KeyError?

Why don't we change all of this to just something like:

```python
try:
    raw['effective-directive'] = raw['violated-directive'].split(None, 1)[0]
except (KeyError, IndexError):
    pass
```

This way we just handle all the use cases a bit more clearly imo.

Reply:
I thought `violated-directive` was required, but now I see that's only for the interface schema.

Updated to use the cleaner try ... except :+1: , but left in the check on to skip when `effective-directive` exists since in the Firefox bug Scott Helme was saying:

> When the block was a fallback to default-src I end up with an effective-directive of default-src, which makes filtering in the GUI a pain

I could go either way on that though just wary about breaking the existing behavior.